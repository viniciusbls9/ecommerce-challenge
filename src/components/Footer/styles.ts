import styled from 'styled-components'
import media from 'styled-media-query'

export const FooterWrapper = styled.section`
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 10.5rem;
  background: transparent linear-gradient(180deg, #ff0000 0%, #f91e96 100%) 0%
    0% no-repeat padding-box;
  display: flex;
  align-items: center;
  justify-content: center;

  ${media.greaterThan('medium')`
    height: 13.5rem;
  `}
`
export const FooterContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 40px;
  margin: 0 auto;
`
export const SingleIconContainer = styled.div``

export const NavigationIconImg = styled.img`
  width: 30px;
  height: 30px;
`
