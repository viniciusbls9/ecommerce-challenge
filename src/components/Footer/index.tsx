import mockNavigation from 'mocks/mock-footer.json'
import Link from 'next/link'
import * as S from './styles'

const Footer = () => (
  <S.Wrapper>
    <S.Container>
      {mockNavigation.map((items) => (
        <Link href={items.redirect} key={items.redirect}>
          <S.SingleIconContainer>
            <S.NavigationIconImg src={items.iconPath} />
          </S.SingleIconContainer>
        </Link>
      ))}
    </S.Container>
  </S.Wrapper>
)

export default Footer
