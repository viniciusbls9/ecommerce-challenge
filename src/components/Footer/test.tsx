import { renderWithTheme } from 'utils/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the footer', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
