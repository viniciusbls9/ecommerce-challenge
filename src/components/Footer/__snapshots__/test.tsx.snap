// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the heading 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 10.5rem;
  background: transparent linear-gradient(180deg,#ff0000 0%,#f91e96 100%) 0% 0% no-repeat padding-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
  gap: 40px;
  margin: 0 auto;
}

.c2 {
  width: 30px;
  height: 30px;
}

@media (min-width:768px) {
  .c0 {
    height: 13.5rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class=""
    >
      <img
        class="c2"
        src="images/face.svg"
      />
    </div>
    <div
      class=""
    >
      <img
        class="c2"
        src="images/insta.svg"
      />
    </div>
    <div
      class=""
    >
      <img
        class="c2"
        src="images/twitter.svg"
      />
    </div>
    <div
      class=""
    >
      <img
        class="c2"
        src="images/youtube.svg"
      />
    </div>
  </div>
</section>
`
